sqlalchemy_drill/base.py

```diff
@@ -0,0 +1,399 @@
+from __future__ import absolute_import
+from __future__ import unicode_literals
+from sqlalchemy import exc, pool, types
+from sqlalchemy.engine import default
+from sqlalchemy.sql import compiler
+from sqlalchemy import inspect
+import requests
+from pprint import pprint
+
+try:
+    from sqlalchemy.sql.compiler import SQLCompiler
+except ImportError:
+    from sqlalchemy.sql.compiler import DefaultCompiler as SQLCompiler
+
+_type_map = {
+    'bit': types.BOOLEAN,
+    'BIT': types.BOOLEAN,
+    'bigint': types.BIGINT,
+    'BIGINT': types.BIGINT,
+    'binary': types.LargeBinary,
+    'BINARY': types.LargeBinary,
+    'boolean': types.BOOLEAN,
+    'BOOLEAN': types.BOOLEAN,
+    'date': types.DATE,
+    'DATE': types.DATE,
+    'decimal': types.DECIMAL,
+    'DECIMAL': types.DECIMAL,
+    'double': types.FLOAT,
+    'DOUBLE': types.FLOAT,
+    'int': types.INTEGER,
+    'INT': types.INTEGER,
+    'integer': types.INTEGER,
+    'INTEGER': types.INTEGER,
+    'interval': types.Interval,
+    'INTERVAL': types.Interval,
+    'smallint': types.SMALLINT,
+    'SMALLINT': types.SMALLINT,
+    'timestamp': types.TIMESTAMP,
+    'TIMESTAMP': types.TIMESTAMP,
+    'time': types.TIME,
+    'TIME': types.TIME,
+    'varchar': types.String,
+    'VARCHAR': types.String,
+    'character varying': types.String,
+    'CHARACTER VARYING': types.String,
+    'ANY': types.String,
+    'any': types.String,
+    'map': types.UserDefinedType,
+    'MAP': types.UserDefinedType,
+    'list': types.UserDefinedType,
+    'LIST': types.UserDefinedType
```

Comment:
It seems excessive to have both upper and lower case variants here, especially as calls to this map seem to always be done using `.lower()`.